### PR TITLE
Fix: Overriding user defined labels

### PR DESCRIPTION
**Pull request by varshaprasad96**
This PR fixes an issue where helm's postrenderer
used to override the user defined object labels instead of merging them.

Issue:
Previously, if any manifest in the bundle had labels provided in them, they were overwritten when bundle deployment was created.

### CODE REVIEW

**Comment by varshaprasad96** [comment]
> Could you please add a test that fails without the fix, and passes with it?

Done. This package did not have tests for bundleDeployment controller, I've added one just to cover post renderer for now.  
Will create issue to add more add coverage to the controller.

File: internal/controllers/bundledeployment/bundledeployment_test.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+			labels := renderedPod.GetLabels()
+			Expect(len(labels)).To(BeEquivalentTo(3))
+			Expect(labels).Should(HaveKeyWithValue("testKey", "testValue"))
+			Expect(labels).Should(HaveKeyWithValue("core.rukpak.io/owner-kind", "BundleDeployment"))
+			Expect(labels).Should(HaveKeyWithValue("core.rukpak.io/owner-name", "test-owner"))
+		})
+
+		It("should add default labels to the pod", func() {
+			pod.SetLabels(map[string]string{})
```
Comment (ncdc) [inline internal/controllers/bundledeployment/bundledeployment_test.go:74]:
```suggestion
			pod.SetLabels(nil)
```

is probably a more representative test. Or you could add another test case so we have 1 for nil, 1 for empty map.

But "should add default labels" is already covered by the previous test case, FYI.

**Review by ncdc — APPROVED** [review]

File: internal/controllers/bundledeployment/bundledeployment_test.go (hunk 1 above)
Comment (varshaprasad96) [reply internal/controllers/bundledeployment/bundledeployment_test.go]: I've rearranged the tests. Let me know if this looks better

**Comment by varshaprasad96** [comment]
Looks like `manifests-diff` maybe failing because my branch name as it has a slash, where its trying to copy `rukpak.yaml` and is not able to find the path. Will close this and re-open the PR with a different branch name :/
